File: sources/net.sf.j2s.core/test/dev/j2s/java/lang/StringBuilder.js
Clazz.load (["java.lang.AbstractStringBuilder", "$.Appendable"], "java.lang.StringBuilder", null, function () {
;
(function(){var C$ = Clazz.decorateAsClass (function () {
Clazz.newInstance$ (this, arguments);
}, java.lang, "StringBuilder", AbstractStringBuilder, [Appendable, CharSequence, java.io.Serializable]);

Clazz.newMethod$(C$, '$init$', function () {
}, 1);

Clazz.newMethod$(C$, 'construct', function () {
C$.superClazz.construct.apply(this, []);
C$.$init$.apply(this);
}, 1);

Clazz.newMethod$(C$, 'construct$I', function (capacity) {
C$.superClazz.construct$I.apply(this, [capacity]);
C$.$init$.apply(this);
}, 1);

Clazz.newMethod$(C$, 'construct$CharSequence', function (seq) {
C$.superClazz.construct$S.apply(this, [seq.toString ()]);
C$.$init$.apply(this);
}, 1);

Clazz.newMethod$(C$, 'construct$S', function (str) {
C$.superClazz.construct$S.apply(this, [str]);
C$.$init$.apply(this);
}, 1);

Clazz.newMethod$(C$, 'append$Z', function (b) {
this.append0$S (b ? "true" : "false");
return this;
});

Clazz.newMethod$(C$, 'append$C', function (c) {
this.append0$C (c);
return this;
});

Clazz.newMethod$(C$, 'append$I', function (i) {
this.append0$S (Integer.toString (i));
return this;
});

Clazz.newMethod$(C$, 'append$J', function (lng) {
this.append0$S (Long.toString (lng));
return this;
});

Clazz.newMethod$(C$, 'append$F', function (f) {
this.append0$S (Float.toString (f));
return this;
});

Clazz.newMethod$(C$, 'append$D', function (d) {
this.append0$S (Double.toString (d));
return this;
});

Clazz.newMethod$(C$, 'append$O', function (obj) {
if (obj == null) {
this.appendNull ();
} else {
this.append0$S (obj.toString ());
}return this;
});

Clazz.newMethod$(C$, 'append$S', function (str) {
this.append0$S (str);
return this;
});

Clazz.newMethod$(C$, 'append$StringBuffer', function (sb) {
if (sb == null) {
this.appendNull ();
} else {
this.append0$CA$I$I (sb.getValue (), 0, sb.length ());
}return this;
});

Clazz.newMethod$(C$, 'append$CA', function (ch) {
this.append0$CA (ch);
return this;
});

Clazz.newMethod$(C$, 'append$CA$I$I', function (str, offset, len) {
this.append0$CA$I$I (str, offset, len);
return this;
});

Clazz.newMethod$(C$, 'append$CharSequence', function (csq) {
if (csq == null) {
this.appendNull ();
} else {
this.append0$S (csq.toString ());
}return this;
});

Clazz.newMethod$(C$, 'append$CharSequence$I$I', function (csq, start, end) {
this.append0$CharSequence$I$I (csq, start, end);
return this;
});

Clazz.newMethod$(C$, '$delete$I$I', function (start, end) {
this.delete0$I$I (start, end);
return this;
});

Clazz.newMethod$(C$, 'deleteCharAt$I', function (index) {
this.deleteCharAt0$I (index);
return this;
});

Clazz.newMethod$(C$, 'insert$I$Z', function (offset, b) {
this.insert0$I$S (offset, b ? "true" : "false");
return this;
});

Clazz.newMethod$(C$, 'insert$I$C', function (offset, c) {
this.insert0$I$C (offset, c);
return this;
});

Clazz.newMethod$(C$, 'insert$I$I', function (offset, i) {
this.insert0$I$S (offset, Integer.toString (i));
return this;
});

Clazz.newMethod$(C$, 'insert$I$J', function (offset, l) {
this.insert0$I$S (offset, Long.toString (l));
return this;
});

Clazz.newMethod$(C$, 'insert$I$F', function (offset, f) {
this.insert0$I$S (offset, Float.toString (f));
return this;
});

Clazz.newMethod$(C$, 'insert$I$D', function (offset, d) {
this.insert0$I$S (offset, Double.toString (d));
return this;
});

Clazz.newMethod$(C$, 'insert$I$O', function (offset, obj) {
this.insert0$I$S (offset, obj == null ? "null" : obj.toString ());
return this;
});

Clazz.newMethod$(C$, 'insert$I$S', function (offset, str) {
this.insert0$I$S (offset, str);
return this;
});

Clazz.newMethod$(C$, 'insert$I$CA', function (offset, ch) {
this.insert0$I$CA (offset, ch);
return this;
});

Clazz.newMethod$(C$, 'insert$I$CA$I$I', function (offset, str, strOffset, strLen) {
this.insert0$I$CA$I$I (offset, str, strOffset, strLen);
return this;
});

Clazz.newMethod$(C$, 'insert$I$CharSequence', function (offset, s) {
this.insert0$I$S (offset, s == null ? "null" : s.toString ());
return this;
});

Clazz.newMethod$(C$, 'insert$I$CharSequence$I$I', function (offset, s, start, end) {
this.insert0$I$CharSequence$I$I (offset, s, start, end);
return this;
});

Clazz.newMethod$(C$, 'replace$I$I$S', function (start, end, str) {
this.replace0$I$I$S (start, end, str);
return this;
});

Clazz.newMethod$(C$, 'reverse', function () {
this.reverse0 ();
return this;
});

Clazz.newMethod$(C$, 'toString', function () {
return C$.superClazz.prototype.toString.apply(this, arguments);
});

Clazz.newMethod$(C$, 'readObject$java_io_ObjectInputStream', function ($in) {
$in.defaultReadObject ();
var count = $in.readInt ();
var value = $in.readObject ();
this.set$CA$I (value, count);
});

Clazz.newMethod$(C$, 'writeObject$java_io_ObjectOutputStream', function (out) {
out.defaultWriteObject ();
out.writeInt$I (this.length ());
out.writeObject$O (this.getValue ());
});
})()
});

//Created 2017-08-17 10:33:14

File: sources/net.sf.j2s.java.core/src/java/lang/StringBuilder.java
/*
 * Copyright (c) 2003, 2013, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

package java.lang;


/**
 * A mutable sequence of characters.  This class provides an API compatible
 * with {@code StringBuffer}, but with no guarantee of synchronization.
 * This class is designed for use as a drop-in replacement for
 * {@code StringBuffer} in places where the string buffer was being
 * used by a single thread (as is generally the case).   Where possible,
 * it is recommended that this class be used in preference to
 * {@code StringBuffer} as it will be faster under most implementations.
 *
 * <p>The principal operations on a {@code StringBuilder} are the
 * {@code append} and {@code insert} methods, which are
 * overloaded so as to accept data of any type. Each effectively
 * converts a given datum to a string and then appends or inserts the
 * characters of that string to the string builder. The
 * {@code append} method always adds these characters at the end
 * of the builder; the {@code insert} method adds the characters at
 * a specified point.
 * <p>
 * For example, if {@code z} refers to a string builder object
 * whose current contents are "{@code start}", then
 * the method call {@code z.append("le")} would cause the string
 * builder to contain "{@code startle}", whereas
 * {@code z.insert(4, "le")} would alter the string builder to
 * contain "{@code starlet}".
 * <p>
 * In general, if sb refers to an instance of a {@code StringBuilder},
 * then {@code sb.append(x)} has the same effect as
 * {@code sb.insert(sb.length(), x)}.
 * <p>
 * Every string builder has a capacity. As long as the length of the
 * character sequence contained in the string builder does not exceed
 * the capacity, it is not necessary to allocate a new internal
 * buffer. If the internal buffer overflows, it is automatically made larger.
 *
 * <p>Instances of {@code StringBuilder} are not safe for
 * use by multiple threads. If such synchronization is required then it is
 * recommended that {@link java.lang.StringBuffer} be used.
 *
 * <p>Unless otherwise noted, passing a {@code null} argument to a constructor
 * or method in this class will cause a {@link NullPointerException} to be
 * thrown.
 *
 * @author      Michael McCloskey
 * @see         java.lang.StringBuffer
 * @see         java.lang.String
 * @since       1.5
 */
public final class StringBuilder
    extends AbstractStringBuilder
    implements java.io.Serializable, CharSequence
{

    /** use serialVersionUID for interoperability */
    static final long serialVersionUID = 4383685877147921099L;

    /**
     * Constructs a string builder with no characters in it and an
     * initial capacity of 16 characters.
     */
    public StringBuilder() {
        super(16);
    }

    /**
     * Constructs a string builder with no characters in it and an
     * initial capacity specified by the {@code capacity} argument.
     *
     * @param      capacity  the initial capacity.
     * @throws     NegativeArraySizeException  if the {@code capacity}
     *               argument is less than {@code 0}.
     */
    public StringBuilder(int capacity) {
        super(capacity);
    }

    /**
     * Constructs a string builder initialized to the contents of the
     * specified string. The initial capacity of the string builder is
     * {@code 16} plus the length of the string argument.
     *
     * @param   str   the initial contents of the buffer.
     */
    public StringBuilder(String str) {
        super(str.length() + 16);
        append(str);
    }

    /**
     * Constructs a string builder that contains the same characters
     * as the specified {@code CharSequence}. The initial capacity of
     * the string builder is {@code 16} plus the length of the
     * {@code CharSequence} argument.
     *
     * @param      seq   the sequence to copy.
     */
    public StringBuilder(CharSequence seq) {
        this(seq.length() + 16);
        append(seq);
    }

    @Override
    public StringBuilder append(Object obj) {
        return (StringBuilder) super.append(obj);
    }

    @Override
    public StringBuilder append(String str) {
        return (StringBuilder) super.append(str);
        
    }

    /**
     * Appends the specified {@code StringBuffer} to this sequence.
     * <p>
     * The characters of the {@code StringBuffer} argument are appended,
     * in order, to this sequence, increasing the
     * length of this sequence by the length of the argument.
     * If {@code sb} is {@code null}, then the four characters
     * {@code "null"} are appended to this sequence.
     * <p>
     * Let <i>n</i> be the length of this character sequence just prior to
     * execution of the {@code append} method. Then the character at index
     * <i>k</i> in the new character sequence is equal to the character at
     * index <i>k</i> in the old character sequence, if <i>k</i> is less than
     * <i>n</i>; otherwise, it is equal to the character at index <i>k-n</i>
     * in the argument {@code sb}.
     *
     * @param   sb   the {@code StringBuffer} to append.
     * @return  a reference to this object.
     */
    @Override
	public StringBuilder append(StringBuffer sb) {
    	return (StringBuilder) super.append(sb);
        
    }

    @Override
    public StringBuilder append(CharSequence s) {
    	return (StringBuilder) super.append(s);
        
    }

    /**
     * @throws     IndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder append(CharSequence s, int start, int end) {
        return (StringBuilder) super.append(s, start, end);
        
    }

    @Override
    public StringBuilder append(char[] str) {
        return (StringBuilder) super.append(str);
        
    }

    /**
     * @throws IndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder append(char[] str, int offset, int len) {
        return (StringBuilder) super.append(str, offset, len);
        
    }

    @Override
    public StringBuilder append(boolean b) {
        return (StringBuilder) super.append(b);
        
    }

    @Override
    public StringBuilder append(char c) {
        return (StringBuilder) super.append(c);
        
    }

    @Override
    public StringBuilder append(int i) {
        return (StringBuilder) super.append( i);
        
    }

    @Override
    public StringBuilder append(long lng) {
        return (StringBuilder) super.append( lng);
        
    }

    @Override
    public StringBuilder append(float f) {
        return (StringBuilder) super.append( f);
        
    }

    @Override
    public StringBuilder append(double d) {
        return (StringBuilder) super.append( d);
        
    }

    /**
     * @since 1.5
     */
    @Override
    public StringBuilder appendCodePoint(int codePoint) {
        return (StringBuilder) super.appendCodePoint(codePoint);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder delete(int start, int end) {
        return (StringBuilder) super.delete(start, end);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder deleteCharAt(int index) {
        return (StringBuilder) super.deleteCharAt(index);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder replace(int start, int end, String str) {
        return (StringBuilder) super.replace(start, end, str);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int index, char[] str, int offset,
                                int len)
    {
        return (StringBuilder) super.insert(index, str, offset, len);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, Object obj) {
            return (StringBuilder) super.insert(offset, obj);
            
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, String str) {
        return (StringBuilder) super.insert(offset, str);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, char[] str) {
        return (StringBuilder) super.insert(offset, str);
        
    }

    /**
     * @throws IndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int dstOffset, CharSequence s) {
            return (StringBuilder) super.insert(dstOffset, s);
            
    }

    /**
     * @throws IndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int dstOffset, CharSequence s,
                                int start, int end)
    {
        return (StringBuilder) super.insert(dstOffset, s, start, end);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, boolean b) {
        return (StringBuilder) super.insert(offset, b);
        
    }

    /**
     * @throws IndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, char c) {
        return (StringBuilder) super.insert(offset, c);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, int i) {
        return (StringBuilder) super.insert(offset, i);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, long l) {
        return (StringBuilder) super.insert(offset, l);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, float f) {
        return (StringBuilder) super.insert(offset, f);
        
    }

    /**
     * @throws StringIndexOutOfBoundsException {@inheritDoc}
     */
    @Override
    public StringBuilder insert(int offset, double d) {
        return (StringBuilder) super.insert(offset, d);
        
    }

    @Override
    public int indexOf(String str) {
        return super.indexOf(str);
    }

    @Override
    public int indexOf(String str, int fromIndex) {
        return super.indexOf(str, fromIndex);
    }

    @Override
    public int lastIndexOf(String str) {
        return super.lastIndexOf(str);
    }

    @Override
    public int lastIndexOf(String str, int fromIndex) {
        return super.lastIndexOf(str, fromIndex);
    }

    @Override
    public StringBuilder reverse() {
    	return (StringBuilder) super.reverse();
        
    }

    @Override
    public String toString() {
    	return 秘s;
//        // Create a copy, don't share the array
//        return new String(value, 0, count);
    }
//
//    /**
//     * Save the state of the {@code StringBuilder} instance to a stream
//     * (that is, serialize it).
//     *
//     * @serialData the number of characters currently stored in the string
//     *             builder ({@code int}), followed by the characters in the
//     *             string builder ({@code char[]}).   The length of the
//     *             {@code char} array may be greater than the number of
//     *             characters currently stored in the string builder, in which
//     *             case extra characters are ignored.
//     */
//    private void writeObject(java.io.ObjectOutputStream s)
//        throws java.io.IOException {
//        s.defaultWriteObject();
//        s.writeInt(count);
//        s.writeObject(value);
//    }
//
//    /**
//     * readObject is called to restore the state of the StringBuffer from
//     * a stream.
//     */
//    private void readObject(java.io.ObjectInputStream s)
//        throws java.io.IOException, ClassNotFoundException {
//        s.defaultReadObject();
//        count = s.readInt();
//        value = (char[]) s.readObject();
//    }

}
